conn.go

```diff
@@ -754,24 +772,24 @@ func (c *Conn) connWriter() {
 		select {
 		// frame write request
 		case env := <-c.txFrame:
-			timeout, ctxErr := c.getWriteTimeout(env.Ctx)
+			timeout, ctxErr := c.getWriteTimeout(env.FrameCtx.Ctx)
 			if ctxErr != nil {
 				debug.Log(1, "TX (connWriter %p) getWriteTimeout: %s: %s", c, ctxErr.Error(), env.Frame)
-				if env.Sent != nil {
-					env.Sent <- ctxErr
+				if env.FrameCtx.CtxErrSem != nil {
```

Comment:
It seems like you could cut this down to a single chan, and use the nil/non-nil state of 'Err' to determine it's failed.

Which is a data structure I wish I had - basically, context with an arbitrary 'err' value.

As it is I think it all makes sense, but having two channels might not be needed.

Reply:
You might be correct.  Let me try...

Reply:
Done